src/wallets/hardware/bcvault/index.js

```diff
@@ -0,0 +1,155 @@
+
+import {
+  getBufferFromHex,
+  sanitizeHex,
+  getSignTransactionObject,
+  calculateChainIdFromV
+} from '../../utils';
+
+const NEED_PASSWORD = false;
+
+class BCVault {
+  constructor() {
+    this.identifier = bcVault;
+    this.isHardware = true;
+    this.needPassword = NEED_PASSWORD;
+    this.selectedAddress = '';
+    this.bcWallet = new bc.BCJS(() => {});
+    this.deviceNumber = null;
+    this.bcWalletType = bc.WalletType.ethereum;
+  }
+
+  async init() {
+    // fetch devices
+    this.deviceNumber = await this.bcWallet.getDevices().catch(errorHandler);
+    // get wallet of first device and password
+    // not sure if we want the users to pass this as a parameter or ask user
+    // for which wallet to use
+    await this.bcWallet
+      .EnterGlobalPin(this.deviceNumber[0], this.bcWalletType)
+      .catch(errorHandler);
+    const walletAddresses = await this.bcWallet.getBatchWalletDetails(
+      this.deviceNumber[0],
+      [bc.WalletType.ethereum]
+    );
+
+    return walletAddresses;
+  }
+
+  getAccount(address) {
+    this.selectedAddress = address;
+    const path = null;
+    const publickey = address;
+    const txSigner = async tx => {
+      if (store.state.main.network.type.chainID === 1) {
+        delete tx['from'];
+        tx['from'] = address;
+        tx = new Transaction(tx, {
+          common: commonGenerator(store.state.main.network)
+        });
+        const newTx = {};
+        newTx['feeCount'] = new BigNumber(
+          bufferToHex(tx['gasLimit'])
+        ).toNumber();
+        newTx['feePrice'] = new BigNumber(
+          bufferToHex(tx['gasPrice'])
+        ).toString();
+        newTx['amount'] =
+          new BigNumber(bufferToHex(tx['value'])).toNumber() || 0;
+        newTx['contractData'] = bufferToHex(tx['data']);
+        newTx['to'] = bufferToHex(tx['to']);
+        newTx['from'] = this.selectedAddress;
+        if (tx.hasOwnProperty('nonce')) {
+          newTx['advanced'] = {
+            eth: {
+              nonce:
+                bufferToHex(tx['nonce']) === '0x'
+                  ? 0
+                  : new BigNumber(bufferToHex(tx['nonce'])).toNumber()
+            }
+          };
+        }
+        const networkId = tx.getChainId();
+        const result = await this.bcWallet
+          .GenerateTransaction(
+            this.deviceNumber[0],
+            this.bcWalletType,
+            newTx,
+            false
+          )
+          .catch(errorHandler);
+        if (result) {
+          const resultTx = new Transaction(result);
+          tx.v = getBufferFromHex(sanitizeHex(resultTx.v.toString('hex')));
+          tx.r = getBufferFromHex(sanitizeHex(resultTx.r.toString('hex')));
+          tx.s = getBufferFromHex(sanitizeHex(resultTx.s.toString('hex')));
+          const signedChainId = calculateChainIdFromV(tx.v);
+          if (signedChainId !== networkId)
+            throw new Error(
+              'Invalid networkId signature returned. Expected: ' +
```

Comment:
this should be passed through i18n

Reply:
This is fixed in coolwallet branch #2087